ncl/lib.ncl

```diff
@@ -178,19 +178,19 @@
       filter_mergeable | {_: Dyn} -> Array FieldDescriptor -> Array FieldDescriptor
         = fun r => array.filter (fun field => mergeable r field),
 
-      elaborate | Array Str -> FieldDescriptor -> ElaboratedField
+      elaborate | Array String -> FieldDescriptor -> ElaboratedField
         = fun prefix field => {
           prio = field.prio,
           path = field.path,
           record = elaborated_record prefix field.path field
         },
 
-      elaborated_record | Array Str -> Array Str -> FieldDescriptor -> {_: Dyn}
+      elaborated_record | Array String -> Array String -> FieldDescriptor -> {_: Dyn}
         = fun prefix acc field =>
-          let head = array.head acc
+          let head = array.first acc
           in if array.length acc > 1
-          then { "%{head}" = elaborated_record prefix (array.tail acc) field, .. }
-          else { "%{head}" | force = provider_computed (prefix @ field.path), .. },
+          then { "%{head}" = elaborated_record prefix (array.drop_first acc) field, .. }
+          else { "%{head}" | (fun l v => provider_computed (prefix @ field.path)) | default = {}, .. },
```

Comment:
Is it generated code? I personally think that inline contracts like that are really...surprising :cold_sweat: I would store them in a variable instead, even if local and used only once

```suggestion
          else { "%{head}" | (fun l v => provider_computed (prefix @ field.path)) | default = {}, .. },
```

Reply:
Well it ends up in generated code :D But this is a really awful workaround for merging not being idempotent. Once that is fixed for arrays, I'd revert this change.